splash/render_options.py

```diff
@@ -51,30 +57,35 @@ def fromrequest(cls, request, max_timeout):
                 (name, value)
                 for name, values in request.requestHeaders.getAllRawHeaders()
                 for value in values
-            ]
-            data.setdefault('headers', headers)
-            data.setdefault('http_method', request.method)
+                ]
+            data.setdefault(b'headers', headers)
+            data.setdefault(b'http_method', request.method)
 
             request.content.seek(0)
-            data.setdefault('body', request.content.read())
+            data.setdefault(b'body', request.content.read())
             request.content.seek(0)
 
-        data['uid'] = id(request)
+        data[b'uid'] = id(request)
         return cls(data, max_timeout)
 
     def get(self, name, default=_REQUIRED, type=str, range=None):
         value = self.data.get(name)
         if value is not None:
             if type is not None:
-                if type is str and isinstance(value, unicode):
-                    value = value.encode('utf8')
+                if type is str and isinstance(value, six.text_type):
+                    value = value.encode('utf-8')
```

Comment:
What about making RenderOptions work on unicode, i.e. accepting unicode and returning unicode? HTTP resources can be in charge of decoding/encoding data.